open_spiel/games/sheriff.cc

```diff
@@ -0,0 +1,392 @@
+    // - it is the *sheriff's* turn. The sheriff can decide to say they will
+    //   _not_ inspect (action id: 0), or that they _will_ inspect the cargo
+    //   (action id: 1).
+    const Player player = CurrentPlayer();
+
+    if (player == kSMUGGLER) {
+      action_ids.reserve(conf.max_bribe + 1);
+      for (uint32_t bribe = 0; bribe <= conf.max_bribe; ++bribe) {
+        action_ids.push_back(sheriff_game_->SerializeBribe(bribe));
+      }
+    } else {
+      action_ids = {sheriff_game_->SerializeInspectionFeedback(false),
+                    sheriff_game_->SerializeInspectionFeedback(true)};
+    }
+  }
+
+  return action_ids;
+}
+
+std::string SheriffState::ActionToString(Player player,
+                                         Action action_id) const {
+  return sheriff_game_->ActionToString(player, action_id);
+}
+
+std::string SheriffState::ToString() const {
+  if (!num_illegal_items_) {
+    return "Initial game state (smuggler hasn't decided the number of illegal "
+           "cargo items yet)";
+  } else {
+    std::string state_str;
+
+    absl::StrAppend(&state_str,
+                    "Num illegal items in cargo: ", *num_illegal_items_, "\n");
+    absl::StrAppend(&state_str, "Bribes  : ");
+    StrAppendVector(&state_str, bribes_);
+    absl::StrAppend(&state_str, "\nFeedback: ");
+    StrAppendVector(&state_str, inspection_feedback_);
+
+    return state_str;
+  }
+}
+
+bool SheriffState::IsTerminal() const {
+  return CurrentPlayer() == kTerminalPlayerId;
+}
+
+std::vector<double> SheriffState::Returns() const {
+  if (!IsTerminal()) {
+    return {0.0, 0.0};
+  } else {
+    SPIEL_CHECK_EQ(inspection_feedback_.size(), bribes_.size());
+    SPIEL_CHECK_GT(inspection_feedback_.size(), 0);
+    SPIEL_CHECK_TRUE(num_illegal_items_);
+
+    const uint32_t num_illegal_items = *num_illegal_items_;
+    const uint32_t bribe = bribes_.back();
+    const bool sheriff_inspects = inspection_feedback_.back();
+    const SheriffGame::SheriffGameConfiguration& conf = sheriff_game_->conf;
+
+    if (sheriff_inspects) {
+      if (num_illegal_items > 0) {
+        // The smuggler was caught red-handed.
+        return {-static_cast<double>(num_illegal_items) * conf.item_penalty,
+                static_cast<double>(num_illegal_items) * conf.item_penalty};
+      } else {
+        // The sheriff must pay up for inspecting a legal cargo.
+        return {conf.sheriff_penalty, -conf.sheriff_penalty};
+      }
+    } else {
+      return {static_cast<double>(num_illegal_items) * conf.item_value - bribe,
+              bribe};
+    }
+  }
+}
+
+std::unique_ptr<State> SheriffState::Clone() const {
+  return std::make_unique<SheriffState>(*this);
+}
+
+std::string SheriffState::InformationStateString(Player player) const {
+  SPIEL_CHECK_TRUE(player >= 0 && player < NumPlayers());
+
+  std::string infostring = absl::StrCat("T=", MoveNumber(), " ");
+  if (player == kSMUGGLER) {
+    infostring = "num_illegal_items:";
```

Comment:
The move number is ignored for the smuggler, is that on purpose?

Reply:
Ah, good catch! That was not done on purpose 😄